Source/WebKit/WebProcess/GPU/media/MediaPlayerPrivateRemote.cpp

```diff
@@ -780,7 +782,8 @@ void MediaPlayerPrivateRemote::remoteVideoTrackConfigurationChanged(TrackPrivate
 #if ENABLE(MEDIA_SOURCE)
 void MediaPlayerPrivateRemote::load(const URL& url, const ContentType& contentType, MediaSourcePrivateClient& client)
 {
-    if (m_remoteEngineIdentifier == MediaPlayerEnums::MediaEngineIdentifier::AVFoundationMSE) {
+    if ((m_remoteEngineIdentifier == MediaPlayerEnums::MediaEngineIdentifier::AVFoundationMSE)
+        || (platformStrategies()->mediaStrategy().mockMediaSourceEnabledInRemoteEngine() && m_remoteEngineIdentifier == MediaPlayerEnums::MediaEngineIdentifier::MockMSE)) {
```

Comment:
OK, in the end this platformStrategies()->mediaStrategy().mockMediaSourceEnabledInRemoteEngine() should be removed, but currently it is required so that we can fix bugs after bugs.
Maybe add a FIXME to remove this check.

Reply:
that's correct. It is still of use to be able to toggle the preference to find regressions if any.